plugins/inputs/postgresql_extensible/postgresql_extensible.go

```diff
@@ -198,40 +193,46 @@ func (p *Postgresql) Gather(acc telegraf.Accumulator) error {
 		sqlQuery += queryAddon
 
 		if p.Query[i].Version <= dbVersion {
-			rows, err := p.DB.Query(sqlQuery)
-			if err != nil {
-				p.Log.Error(err.Error())
-				continue
-			}
+			p.gatherMetricsFromQuery(acc, sqlQuery, p.Query[i].Tagvalue, p.Query[i].Timestamp, measName)
+		}
+	}
+	return nil
+}
 
-			defer rows.Close()
+func (p *Postgresql) gatherMetricsFromQuery(acc telegraf.Accumulator, sqlQuery string, tagValue string, timestamp string, measName string) {
+	var columns []string
 
-			// grab the column information from the result
-			if columns, err = rows.Columns(); err != nil {
-				p.Log.Error(err.Error())
-				continue
-			}
+	rows, err := p.DB.Query(sqlQuery)
+	if err != nil {
+		p.Log.Error(err.Error())
```

Comment:
Maybe `acc.AddError` here to get error statistics. More similar locations below.

Reply:
Good catch, changed!